pkg/apis/messaging/v1alpha1/nats_jetstream_types.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2021 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	eventingduckv1 "knative.dev/eventing/pkg/apis/duck/v1"
+	"knative.dev/pkg/apis"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	"knative.dev/pkg/kmeta"
+)
+
+// +genclient
+// +genreconciler
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// NatsJetStreamChannel is a resource representing a NATS JetStream Channel.
```

Comment:
I had thought that the only thing we needed to modify was the actual dispatcher code? I'm curious why we need to create entirely new channel type because the actual type does not change. Just wondering if we could just make it a configuration option of which kind of dispatcher we use for the dataplane? Or are there actual control plane changes that come with the move to jetstream? Just jotting this down while reading the code.

Reply:
I think it is necessary to join a new channel. For example, users can specify `streamName`(hasn't been added to the `natsjetstreamchannel`) in the `channel` and other features（maybe more than I can see now） that NATSS doesn’t have.  
what do you thank ？@devguyio

Reply:
Sure thing, I don't have enough knowledge about natss or jetstream to say one way or another, I had just (mis)understood from the previous conversation that it was a change in the libraries only, my bad :)